apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/sidebar-modal-opener.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+import { Button, Modal } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { PageTemplatesPlugin } from '../index';
+import TemplateSelectorItem from './template-selector-item';
+import replacePlaceholders from '../utils/replace-placeholders';
+/* eslint-enable import/no-extraneous-dependencies */
+
+class SidebarModalOpener extends Component {
+	state = {
+		isTemplateModalOpen: false,
+		isWarningOpen: false,
+	};
+
+	toggleTemplateModal = () => {
+		this.setState( {
+			isTemplateModalOpen: ! this.state.isTemplateModalOpen,
+			isWarningOpen: false,
+		} );
+	};
+
+	toggleWarningModal = () => {
+		this.setState( { isWarningOpen: ! this.state.isWarningOpen } );
+	};
+
+	getLastTemplateUsed = () => {
+		if ( ! this.props.templateUsedSlug || this.props.templateUsedSlug === 'blank' ) {
+			return this.props.templates[ 1 ];
+		}
+		return this.props.templates.filter( temp => temp.slug === this.props.templateUsedSlug )[ 0 ];
```

Comment:
I think [Array.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) might be a better fit here :) 

```suggestion
		return this.props.templates.find( temp => temp.slug === this.props.templateUsedSlug );
```

Reply:
LOL @ self...  agreed and updated.